public/app/plugins/panel/graph/align_yaxes.ts

```diff
@@ -0,0 +1,136 @@
+import _ from 'lodash';
+
+/**
+ * To align two Y axes by Y level
+ * @param yaxis data [{min: min_y1, min: max_y1}, {min: min_y2, max: max_y2}]
+ * @param align Y level
+ */
+export function alignYLevel(yaxis, alignLevel) {
+  moveLevelToZero(yaxis, alignLevel);
+
+  expandStuckValues(yaxis);
+
+  // one of graphs on zero
+  var zero = yaxis[0].min === 0 || yaxis[1].min === 0 || yaxis[0].max === 0 || yaxis[1].max === 0;
+
+  var oneSide = checkOneSide(yaxis);
+
+  if (zero && oneSide) {
+    yaxis[0].min = yaxis[0].max > 0 ? 0 : yaxis[0].min;
+    yaxis[0].max = yaxis[0].max > 0 ? yaxis[0].max : 0;
+    yaxis[1].min = yaxis[1].max > 0 ? 0 : yaxis[1].min;
+    yaxis[1].max = yaxis[1].max > 0 ? yaxis[1].max : 0;
+  } else {
+    // on the opposite sides with respect to zero
+    if ((yaxis[0].min >= 0 && yaxis[1].max <= 0) || (yaxis[0].max <= 0 && yaxis[1].min >= 0)) {
```

Comment:
I still find this code and below could be easier to understand if divided into smaller functions. Possible?